tests/proof_systems/t_r1cs_parser.nim

```diff
@@ -0,0 +1,83 @@
+import
+  std/[os, unittest, strutils],
+  ../../constantine/proof_systems/constraint_systems/r1cs_circom_parser
+
+
+const TestDir = currentSourcePath.rsplit(DirSep, 1)[0]
+suite "R1CS binary file parser":
+
+  test "Parse basic example R1CS file":
+    # Note: The example test file used here is from:
+    # https://github.com/iden3/r1csfile/tree/master/test/testutils
+    const path = TestDir / "r1cs_test_files/example.r1cs"
+    let r1cs = parseR1csFile(path)
+
+    ## XXX: On 32bit systems `seq[BaseType]` should have twice as many elements
+    ## in the `expABCs` below as `value` arguments!
```

Comment:
Given that when converting to `Factor` we aren't even validating the elements, see:
- https://github.com/mratsim/constantine/pull/388/files#diff-f275b0ff9384fffa4fb5adc6a027fc7deaf68890029efd3a59a0911c047ee3adR36
- https://github.com/mratsim/constantine/pull/388/files#diff-f275b0ff9384fffa4fb5adc6a027fc7deaf68890029efd3a59a0911c047ee3adR128

We might as well have Factor use a `seq[byte]` to be uint32/uint64 agnostic. And do the conversion in a later pass, along with validation.